### ISSUE TITLE
`DatabaseName must begin with a letter and contain only alphanumeric characters.` error in Function + Posgres + Drizzle ORM setup

### ISSUE THREAD

**Issue by frixaco** [sst/ion#56]
Simple Lambda function linked to a Postgres database giving following error when running `sst dev`:

```
❯ pnpm dev --verbose

> test-sst-ion@ dev /Users/frixaco/personal/test-sst-ion
> AWS_PROFILE=<> dev "--verbose"

time=2024-03-24T18:28:35.446+05:00 level=INFO msg="checking for bun" path="/Users/frixaco/Library/Application Support/sst/bin/bun"
time=2024-03-24T18:28:35.447+05:00 level=INFO msg=args args=[] length=0
time=2024-03-24T18:28:35.447+05:00 level=INFO msg="no existing server found, starting new one"
time=2024-03-24T18:28:35.448+05:00 level=INFO msg="waiting for server to start"
SST ❍ ion 0.0.197  ready!

➜  App:        test-sst-ion
   Stage:      frixaco
   Console:    https://console.sst.dev/local/test-sst-ion/frixaco

~  Deploying

|  Creating    TestSSTDatabase sst:aws:Postgres → TestSSTDatabaseCluster aws:rds:Cluster
|  Error       TestSSTDatabase sst:aws:Postgres → TestSSTDatabaseCluster aws:rds:Cluster
|  DatabaseName must begin with a letter and contain only alphanumeric characters.
|  94fbd721-92ae-4528-afe3-404bf911e3a6

×  Failed
   TestSSTDatabase sst:aws:Postgres → TestSSTDatabaseCluster aws:rds:Cluster
   DatabaseName must begin with a letter and contain only alphanumeric characters.
   94fbd721-92ae-4528-afe3-404bf911e3a6
```

`sst.config.ts`:
```ts

export default $config({
  app(input) {
    return {
      name: "test-sst-ion",
      removal: input?.stage === "production" ? "retain" : "remove",
      home: "aws",
    };
  },
  async run() {
    const database = new sst.aws.Postgres("TestSSTDatabase");

    new sst.aws.Function("TestSSTFunction", {
      link: [database],
      handler: "src/lambda.handler",
      url: {
        authorization: "none",
        cors: {
          allowMethods: ["GET", "POST"],
          allowOrigins: ["*"],
        },
      },
   });
  },
});

```

Full repro: https://github.com/frixaco/test-sst-ion

Is it a bug on Ion or did I misconfigure something?

Note: The main goal is to setup Next.js app with automatic DB migrations when using Drizzle ORM when deploying to different environments. After some digging on Discord, I found a link to an example which uses `Script` construct ([config file](https://github.com/sst/sst/blob/master/examples/nextjs-rds-drizzle/stacks/Default.ts)). Ion doesn't have `Script` construct, however, Dax mentioned that it might not be needed. My idea is to create a simple Function bundled with migration files and send a POST request to that function that triggers `migrate` function ([config file](https://github.com/frixaco/test-sst-ion/blob/038499d0de2df106e45f63a12d8a1c80269556da/sst.config.ts#L34))

<sub>[ION-237](https://linear.app/sst-dev/issue/ION-237/databasename-must-begin-with-a-letter-and-contain-only-alphanumeric)</sub>

**Comment by frixaco** [comment]
Maybe it's a linking issue? Running `pnpm db:gen` gives following error:

```
❯ pnpm db:gen --verbose

> test-sst-ion@ db:gen /Users/frixaco/personal/test-sst-ion
> AWS_PROFILE=default sst shell drizzle-kit generate:pg "--verbose"

time=2024-03-24T19:41:43.422+05:00 level=INFO msg="checking for bun" path="/Users/frixaco/Library/Application Support/sst/bin/bun"
time=2024-03-24T19:41:43.422+05:00 level=INFO msg="initializing project" version=0.0.197
time=2024-03-24T19:41:43.423+05:00 level=INFO msg="esbuild building"
time=2024-03-24T19:41:43.425+05:00 level=INFO msg="esbuild built" outfile=/Users/frixaco/personal/test-sst-ion/.sst/eval/eval-1711291303423.mjs
time=2024-03-24T19:41:43.425+05:00 level=INFO msg="evaluating config"
time=2024-03-24T19:41:43.451+05:00 level=INFO msg="config evaluated"
time=2024-03-24T19:41:43.451+05:00 level=INFO msg="checking platform"
time=2024-03-24T19:41:43.452+05:00 level=INFO msg="credentials found"
time=2024-03-24T19:41:43.452+05:00 level=INFO msg="fetching bootstrap"
time=2024-03-24T19:41:45.026+05:00 level=INFO msg="found existing bootstrap" data="{\"version\":1,\"asset\":\"sst-asset-runmwdcehbtz\",\"state\":\"sst-state-runmwdcehbtz\"}"
time=2024-03-24T19:41:45.027+05:00 level=INFO msg="loaded config" app=test-sst-ion stage=frixaco
time=2024-03-24T19:41:45.027+05:00 level=INFO msg="INFO getting data key=link app=test-sst-ion stage=frixaco"
drizzle-kit: v0.20.14
drizzle-orm: v0.30.4

No config path provided, using default 'drizzle.config.ts'
Reading config file '/Users/frixaco/personal/test-sst-ion/drizzle.config.ts'
Error: "TestSSTDatabase" is not linked
    at Object.get (/Users/frixaco/personal/test-sst-ion/node_modules/.pnpm/sst@3.0.2/node_modules/sst/dist/resource.js:15:15)
    at Object.<anonymous> (/Users/frixaco/personal/test-sst-ion/database.ts:19:22)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._compile (/Users/frixaco/personal/test-sst-ion/node_modules/.pnpm/drizzle-kit@0.20.14/node_modules/drizzle-kit/bin.cjs:8644:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Object.newLoader [as .ts] (/Users/frixaco/personal/test-sst-ion/node_modules/.pnpm/drizzle-kit@0.20.14/node_modules/drizzle-kit/bin.cjs:8648:13)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Module.require (node:internal/modules/cjs/loader:1235:19)
    at require (node:internal/modules/helpers:176:18)
```

**Comment by fwang** [comment]
@frixaco what's the name of the app defined in `sst.config.ts`?

**Comment by frixaco** [comment]
@fwang Thanks for a response. I'm not sure I understand what do you mean by "app". If you are asking about whether it's a Next.js/Remix app, then no. It's just a Function linked to a database (wanted to make bundling migration files with lambda handler work in a Function).

Let me know if you need more details.

**Comment by thdxr** [comment]
we mean what's at the top of your sst.config.ts, the part you cut out

when you don't specify a default database name to the postgres component we use the app name as the database name, seems like yours is not valid

**Comment by frixaco** [comment]
@thdxr Thanks for clarifying. 
It is `test-sst-ion` 🤦‍♂️ , it is using the name of root project directory. Sorry, that's my bad for being a bit careless, didn't realize that.
Maybe Ion can auto "fix" the app name when running `sst init`? (I'm down to implement it, though I haven't looked at the source yet)

**Comment by thdxr** [comment]
oh can database names not have hyphens? @fwang should we strip them?

**Comment by jayair** [comment]
Gonna assign to Frank

**Comment by fwang** [comment]
Fixed in 0.0.226
